host/ProtocolGateway.Host.Fabric.FrontEnd/mqtt/MqttCommunicationListener.cs

```diff
@@ -0,0 +1,361 @@
+
+            X509Certificate2 certificate = this.GetServerCertificate(configuration);
+            this.logger.Informational(traceId, this.componentName, "Certificate retrieved.");
+
+            var mqttConfigurationProvider = new ConfigurationProvider<MqttServiceConfiguration>(traceId, this.serviceContext.ServiceName, this.serviceContext.CodePackageActivationContext, this.logger, "Mqtt");
+            mqttConfigurationProvider.OnConfigurationClassChangedEvent += (sender, args) => this.unsupportedConfigurationChangeCallback?.Invoke();
+            mqttConfigurationProvider.OnConfigurationPropertyChangedEvent += (sender, args) => this.unsupportedConfigurationChangeCallback?.Invoke();
+            MqttServiceConfiguration mqttConfiguration = mqttConfigurationProvider.Config;
+            
+            if (mqttConfiguration != null)
+            {
+                var mqttInboundTemplates = new List<string>(mqttConfiguration.MqttInboundTemplates);
+                var mqttOutboundTemplates = new List<string>(mqttConfiguration.MqttOutboundTemplates);
+
+                ISessionStatePersistenceProvider qosSessionProvider = await this.GetQosStateProvider(traceId, mqttConfiguration).ConfigureAwait(false);
+                IQos2StatePersistenceProvider qos2SessionProvider = await this.GetQos2StateProvider(traceId, mqttConfiguration).ConfigureAwait(false);
+
+                this.logger.Informational(traceId, this.componentName, "QOS Providers instantiated.");
+
+                var settingsProvider = new ServiceFabricConfigurationProvider(traceId, this.componentName, this.logger, configuration, iotHubConfiguration, mqttConfiguration);
+                this.bootStrapper = new Bootstrapper(settingsProvider, qosSessionProvider, qos2SessionProvider, mqttInboundTemplates, mqttOutboundTemplates);
+                this.runTask = this.bootStrapper.RunAsync(certificate, threadCount, cancellationToken);
+
+                publishAddress = this.BuildPublishAddress(configuration);
+
+                this.logger.Informational(traceId, this.componentName, "Bootstrapper instantiated.");
+            }
+            else
+            {
+                this.logger.Critical(traceId, this.componentName, "Failed to start endpoint because Mqtt service configuration is missing.");
+            }
+
+            return publishAddress;
+        }
+        #endregion
+        #region QoS Provider
+        /// <summary>
+        /// Retrieves the state provider for QoS 0/1 sessions
+        /// </summary>
+        /// <param name="traceId">A unique identifier used to correlate debugging and diagnostics messages</param>
+        /// <param name="configuration">The gateway configuration</param>
+        /// <returns>A Qos 0/1 persistence  provider if available</returns>
+        async Task<ISessionStatePersistenceProvider> GetQosStateProvider(Guid traceId, MqttServiceConfiguration configuration)
```

Comment:
That's misleading; MQTT Session State consists of list of subscriptions and does not have much to do with QoS 0/1 specifically. how about "GetSessionStateProviderAsync"? Also, pls update xml doc accordingly.